### ISSUE TITLE
Compile a static executable program

### ISSUE THREAD

**Issue by yumi-di** [kahypar/mt-kahypar#159]
Hi,
I am trying to compile a static executable program that can run in different Linux environments,
However, I found it difficult to compile static programs because of the TBB library.
Is there a method for compiling static programs? Or is there any suggestion to run the program in different environments.

thanks!

**Comment by larsgottesbueren** [comment]
Hi,

Currently there is no method to compile our code to a static binary. It seems that the TBB developers strongly advise against it and do not offer a static library themselves. You can try to roll your own with the steps outlined in https://stackoverflow.com/questions/638278/how-to-statically-link-to-tbb. But I'm guessing you found that link and the associated caveats already. This seems to be the easiest option to run in different environments. We're not planning to move away from TBB any time soon. 

Best,
Lars

**Comment by uulm-janbaudisch** [comment]
Hi,
linking `libmtkahypar` statically should work though. I tried it by modifying `lib/CMakeLists.txt` to build a static version of `libmtkahypar` and linking it to my executable, but still linking TBB dynamically.
Would it be acceptable for this project to include the possibility for a static library?

**Comment by larsgottesbueren** [comment]
I think that should be fine. We're happy to take contributions.

**Comment by uulm-janbaudisch** [comment]
Implemented in #161.

**Comment by uulm-janbaudisch** [comment]
Assuming #161 is implemented correctly, it currently leads to the following behaviour:

The dynamic version (`BUILD_SHARED_LIBS=ON`) works fine (tested with `lib/examples/partition_graph.cc`), but using the static version (`BUILD_SHARED_LIBS=OFF`), which builds correctly, it exits with the following error message:

```
Could not load mt_kahypar::CoarseningAlgorithm :  multilevel_coarsener
```

So basically the same code behaves differently when used as a shared or static library.

I am not familiar enough with this project to pin down the issue and I understand that supporting a static version of the library interface is not priority but maybe someone has an idea on what the issue could be.
